Changed date guessing heuristic to use last modification time of posts instead of current time, if no better data is available.

kodfodrasz opened the pull request:
Changed date guessing heuristic to use last modification time of posts instead of current time, if no better data is available.

This is a viable and better replacement as all filesystems track this, and git and other version control tools also keep track of this and set it on files on checkout.

The changes:

 - Instead of DateTime.Now the source file's LastWriteTime is used now
 - The tests were updated
 - Updated System.IO.Abstractions and TestingHelpers to version 2.0.0.116
   to avoid a bug in the System.IO.Abstractions.TestingHelpers library
   making tests to fail
   (see: https://github.com/tathamoddie/System.IO.Abstractions/issues/122 )

laedit commented on src/Pretzel.Tests/Templating/Context/SiteContextGeneratorTests.cs:
```diff
@@ -791,21 +807,25 @@ public void page_with_false_date_in_title()
         public void post_default_values()
         {
-            fileSystem.AddFile(@"C:\TestSite\_posts\SomeFile.md", new MockFileData(@"---
+            var filename = @"C:\TestSite\SomeFile.md";
```
Why the change from Posts to Pages?

kodfodrasz replied:
Because the test failed. The test failed because the Posts get populated from _posts, Pages gets populated from anywhere in the site source. The path in the test vectors was not in the _posts folder (which made me wonder why the tests didn't fail earlier...). Instead of wasting hours, i changed it.

I hope I have not misunderstood something.

laedit replied:
Well the file path was modified from "C:\TestSite\_posts\SomeFile.md" to "C:\TestSite\SomeFile.md" so it's normal that Posts is empty, but why change the file path?

laedit commented on src/Pretzel.Tests/Templating/Jekyll/LiquidEngineTests.cs:
```diff
@@ -787,8 +787,8 @@ public override LiquidEngine Given()
             {
-                FileSystem.AddFile(@"C:\website\_site\BadFormat.md", new MockFileData(OriginalPageContents) { LastWriteTime = new DateTime(2010, 01, 2) });
-                FileSystem.AddFile(@"C:\website\BadFormat.md", new MockFileData(PageContents) { LastWriteTime = new DateTime(2010, 01, 3) });
+                FileSystem.AddFile(@"C:\website\_site\BadFormat.md", new MockFileData(OriginalPageContents) { LastWriteTime = new DateTime(2013, 01, 2) });
```
Why the year modification? Do you wanted to test different year values?

kodfodrasz replied:
I don't know. It may have come from some silly idea. I have cherry picked this commit from a dormant branch I was using since about half a year ago.

laedit replied:
Since it is not used for the tests, could you revert the changes?

laedit commented:
Good one, thanks a lot!
I just have some question about the tests modifications.

kodfodrasz commented:
I have addressed your comments. They were totally rightful. Sorry for them, I have just recently pulled some of my changes together from my sandbox branch, and decided to send them. This PR was only sandbox-quality.

laedit commented:
No problem, thanks again for your work! :smiley:
